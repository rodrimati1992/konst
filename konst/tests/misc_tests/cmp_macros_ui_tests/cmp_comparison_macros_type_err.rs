use konst::cmp;

use core::cmp::Ordering;


struct BadCmp;

cmp::impl_cmp! {
    impl BadCmp;

    pub const fn const_eq(&self, _other: &Self) -> u8 {
        0u8
    }
    pub const fn const_cmp(&self, _other: &Self) -> u8 {
        0u8
    }
}


const _: bool = cmp::const_eq!(0u8, ());
const _: () = cmp::const_eq!(0u8, 0u8);
const _: () = { cmp::const_eq!(BadCmp, BadCmp); };
const _: () = cmp::const_eq!(BadCmp, BadCmp);

const _: bool = cmp::const_ne!(0u8, ());
const _: () = cmp::const_ne!(0u8, 0u8);
const _: () = { cmp::const_ne!(BadCmp, BadCmp); };
const _: () = cmp::const_ne!(BadCmp, BadCmp);

const _: Ordering = cmp::const_cmp!(0u8, ());
const _: () = cmp::const_cmp!(0u8, 0u8);
const _: () = { cmp::const_cmp!(BadCmp, BadCmp); };
const _: () = cmp::const_cmp!(BadCmp, BadCmp);

const _: bool = cmp::const_lt!(0u8, ());
const _: () = cmp::const_lt!(0u8, 0u8);
const _: () = { cmp::const_lt!(BadCmp, BadCmp); };
const _: () = cmp::const_lt!(BadCmp, BadCmp);

const _: bool = cmp::const_le!(0u8, ());
const _: () = cmp::const_le!(0u8, 0u8);
const _: () = { cmp::const_le!(BadCmp, BadCmp); };
const _: () = cmp::const_le!(BadCmp, BadCmp);

const _: bool = cmp::const_gt!(0u8, ());
const _: () = cmp::const_gt!(0u8, 0u8);
const _: () = { cmp::const_gt!(BadCmp, BadCmp); };
const _: () = cmp::const_gt!(BadCmp, BadCmp);

const _: bool = cmp::const_ge!(0u8, ());
const _: () = cmp::const_ge!(0u8, 0u8);
const _: () = { cmp::const_ge!(BadCmp, BadCmp); };
const _: () = cmp::const_ge!(BadCmp, BadCmp);


fn main() {}