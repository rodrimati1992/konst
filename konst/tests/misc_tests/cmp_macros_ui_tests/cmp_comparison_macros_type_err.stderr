error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:20:17
   |
20 | const _: bool = cmp::const_eq!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
note: required by a bound in `__AssertConstCmp`
  --> src/cmp/const_cmp.rs
   |
   | pub struct __AssertConstCmp<'a, T: ConstCmp>{
   |                                    ^^^^^^^^ required by this bound in `__AssertConstCmp`
   = note: this error originates in the macro `cmp::const_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:20:17
   |
20 | const _: bool = cmp::const_eq!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
   = note: required for `()` to implement `ConstCmpUnref`
note: required by a bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
  --> src/cmp/const_cmp.rs
   |
   |     R: ?Sized + ConstCmpUnref<This = T>,
   |                               ^^^^^^^^ required by this bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
...
   |     pub const NEW: Self = Self(PhantomData);
   |               --- required by a bound in this associated constant
   = note: this error originates in the macro `$crate::__coerce_to_cmp2` which comes from the expansion of the macro `cmp::const_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:20:17
   |
20 | const _: bool = cmp::const_eq!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `&u8`, found `&()`
   |                 arguments to this method are incorrect
   |
   = note: expected reference `&u8`
              found reference `&()`
note: method defined here
  --> src/cmp/cmp_wrapper.rs
   |
   | / std_kind_impls! {
   | |     i8, u8,
   | |     i16, u16,
   | |     i32, u32,
...  |
   | |     bool, char,
   | | }
   | |_^
   = note: this error originates in the macro `cmp::const_eq` which comes from the expansion of the macro `std_kind_impls` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:21:15
   |
21 | const _: () = cmp::const_eq!(0u8, 0u8);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
   |
   = note: this error originates in the macro `cmp::const_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:22:17
   |
22 | const _: () = { cmp::const_eq!(BadCmp, BadCmp); };
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `bool`, found `u8`
   |                 expected due to this
   |
   = note: this error originates in the macro `cmp::const_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:23:15
   |
23 | const _: () = cmp::const_eq!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |               |
   |               expected `bool`, found `u8`
   |               expected due to this
   |
   = note: this error originates in the macro `cmp::const_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:23:15
   |
23 | const _: () = cmp::const_eq!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
   |
   = note: this error originates in the macro `cmp::const_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:25:21
   |
25 | const _: Ordering = cmp::const_cmp!(0u8, ());
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
note: required by a bound in `__AssertConstCmp`
  --> src/cmp/const_cmp.rs
   |
   | pub struct __AssertConstCmp<'a, T: ConstCmp>{
   |                                    ^^^^^^^^ required by this bound in `__AssertConstCmp`
   = note: this error originates in the macro `cmp::const_cmp` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:25:21
   |
25 | const _: Ordering = cmp::const_cmp!(0u8, ());
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
   = note: required for `()` to implement `ConstCmpUnref`
note: required by a bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
  --> src/cmp/const_cmp.rs
   |
   |     R: ?Sized + ConstCmpUnref<This = T>,
   |                               ^^^^^^^^ required by this bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
...
   |     pub const NEW: Self = Self(PhantomData);
   |               --- required by a bound in this associated constant
   = note: this error originates in the macro `$crate::__coerce_to_cmp2` which comes from the expansion of the macro `cmp::const_cmp` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:25:21
   |
25 | const _: Ordering = cmp::const_cmp!(0u8, ());
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^
   |                     |
   |                     expected `&u8`, found `&()`
   |                     arguments to this method are incorrect
   |
   = note: expected reference `&u8`
              found reference `&()`
note: method defined here
  --> src/primitive/cmp.rs
   |
   | / __declare_fns_with_docs! {
   | |     (u8, (cmp_u8))
   | |     (u16, (cmp_u16))
   | |     (u32, (cmp_u32))
...  |
   | |     macro = declare_int_cmp_fn!(),
   | | }
   | |_^
   = note: this error originates in the macro `cmp::const_cmp` which comes from the expansion of the macro `__declare_fns_with_docs` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:26:15
   |
26 | const _: () = cmp::const_cmp!(0u8, 0u8);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `Ordering`
   |
   = note: this error originates in the macro `cmp::const_cmp` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:27:17
   |
27 | const _: () = { cmp::const_cmp!(BadCmp, BadCmp); };
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `Ordering`, found `u8`
   |                 expected due to this
   |
   = note: this error originates in the macro `cmp::const_cmp` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:28:15
   |
28 | const _: () = cmp::const_cmp!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |               |
   |               expected `Ordering`, found `u8`
   |               expected due to this
   |
   = note: this error originates in the macro `cmp::const_cmp` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:28:15
   |
28 | const _: () = cmp::const_cmp!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `Ordering`
   |
   = note: this error originates in the macro `cmp::const_cmp` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:30:17
   |
30 | const _: bool = cmp::const_lt!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
note: required by a bound in `__AssertConstCmp`
  --> src/cmp/const_cmp.rs
   |
   | pub struct __AssertConstCmp<'a, T: ConstCmp>{
   |                                    ^^^^^^^^ required by this bound in `__AssertConstCmp`
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_lt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:30:17
   |
30 | const _: bool = cmp::const_lt!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
   = note: required for `()` to implement `ConstCmpUnref`
note: required by a bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
  --> src/cmp/const_cmp.rs
   |
   |     R: ?Sized + ConstCmpUnref<This = T>,
   |                               ^^^^^^^^ required by this bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
...
   |     pub const NEW: Self = Self(PhantomData);
   |               --- required by a bound in this associated constant
   = note: this error originates in the macro `$crate::__coerce_to_cmp2` which comes from the expansion of the macro `cmp::const_lt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:30:17
   |
30 | const _: bool = cmp::const_lt!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `&u8`, found `&()`
   |                 arguments to this method are incorrect
   |
   = note: expected reference `&u8`
              found reference `&()`
note: method defined here
  --> src/primitive/cmp.rs
   |
   | / __declare_fns_with_docs! {
   | |     (u8, (cmp_u8))
   | |     (u16, (cmp_u16))
   | |     (u32, (cmp_u32))
...  |
   | |     macro = declare_int_cmp_fn!(),
   | | }
   | |_^
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `__declare_fns_with_docs` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:31:15
   |
31 | const _: () = cmp::const_lt!(0u8, 0u8);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
   |
   = note: this error originates in the macro `$crate::__::matches` which comes from the expansion of the macro `cmp::const_lt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:32:17
   |
32 | const _: () = { cmp::const_lt!(BadCmp, BadCmp); };
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `Ordering`, found `u8`
   |                 expected due to this
   |
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_lt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:33:15
   |
33 | const _: () = cmp::const_lt!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |               |
   |               expected `Ordering`, found `u8`
   |               expected due to this
   |
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_lt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:33:15
   |
33 | const _: () = cmp::const_lt!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
   |
   = note: this error originates in the macro `$crate::__::matches` which comes from the expansion of the macro `cmp::const_lt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:35:17
   |
35 | const _: bool = cmp::const_le!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
note: required by a bound in `__AssertConstCmp`
  --> src/cmp/const_cmp.rs
   |
   | pub struct __AssertConstCmp<'a, T: ConstCmp>{
   |                                    ^^^^^^^^ required by this bound in `__AssertConstCmp`
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_le` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:35:17
   |
35 | const _: bool = cmp::const_le!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
   = note: required for `()` to implement `ConstCmpUnref`
note: required by a bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
  --> src/cmp/const_cmp.rs
   |
   |     R: ?Sized + ConstCmpUnref<This = T>,
   |                               ^^^^^^^^ required by this bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
...
   |     pub const NEW: Self = Self(PhantomData);
   |               --- required by a bound in this associated constant
   = note: this error originates in the macro `$crate::__coerce_to_cmp2` which comes from the expansion of the macro `cmp::const_le` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:35:17
   |
35 | const _: bool = cmp::const_le!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `&u8`, found `&()`
   |                 arguments to this method are incorrect
   |
   = note: expected reference `&u8`
              found reference `&()`
note: method defined here
  --> src/primitive/cmp.rs
   |
   | / __declare_fns_with_docs! {
   | |     (u8, (cmp_u8))
   | |     (u16, (cmp_u16))
   | |     (u32, (cmp_u32))
...  |
   | |     macro = declare_int_cmp_fn!(),
   | | }
   | |_^
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `__declare_fns_with_docs` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:36:15
   |
36 | const _: () = cmp::const_le!(0u8, 0u8);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
   |
   = note: this error originates in the macro `$crate::__::matches` which comes from the expansion of the macro `cmp::const_le` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:37:17
   |
37 | const _: () = { cmp::const_le!(BadCmp, BadCmp); };
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `Ordering`, found `u8`
   |                 expected due to this
   |
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_le` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:38:15
   |
38 | const _: () = cmp::const_le!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |               |
   |               expected `Ordering`, found `u8`
   |               expected due to this
   |
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_le` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:38:15
   |
38 | const _: () = cmp::const_le!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
   |
   = note: this error originates in the macro `$crate::__::matches` which comes from the expansion of the macro `cmp::const_le` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:40:17
   |
40 | const _: bool = cmp::const_gt!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
note: required by a bound in `__AssertConstCmp`
  --> src/cmp/const_cmp.rs
   |
   | pub struct __AssertConstCmp<'a, T: ConstCmp>{
   |                                    ^^^^^^^^ required by this bound in `__AssertConstCmp`
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_gt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:40:17
   |
40 | const _: bool = cmp::const_gt!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
   = note: required for `()` to implement `ConstCmpUnref`
note: required by a bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
  --> src/cmp/const_cmp.rs
   |
   |     R: ?Sized + ConstCmpUnref<This = T>,
   |                               ^^^^^^^^ required by this bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
...
   |     pub const NEW: Self = Self(PhantomData);
   |               --- required by a bound in this associated constant
   = note: this error originates in the macro `$crate::__coerce_to_cmp2` which comes from the expansion of the macro `cmp::const_gt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:40:17
   |
40 | const _: bool = cmp::const_gt!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `&u8`, found `&()`
   |                 arguments to this method are incorrect
   |
   = note: expected reference `&u8`
              found reference `&()`
note: method defined here
  --> src/primitive/cmp.rs
   |
   | / __declare_fns_with_docs! {
   | |     (u8, (cmp_u8))
   | |     (u16, (cmp_u16))
   | |     (u32, (cmp_u32))
...  |
   | |     macro = declare_int_cmp_fn!(),
   | | }
   | |_^
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `__declare_fns_with_docs` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:41:15
   |
41 | const _: () = cmp::const_gt!(0u8, 0u8);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
   |
   = note: this error originates in the macro `$crate::__::matches` which comes from the expansion of the macro `cmp::const_gt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:42:17
   |
42 | const _: () = { cmp::const_gt!(BadCmp, BadCmp); };
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `Ordering`, found `u8`
   |                 expected due to this
   |
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_gt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:43:15
   |
43 | const _: () = cmp::const_gt!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |               |
   |               expected `Ordering`, found `u8`
   |               expected due to this
   |
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_gt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:43:15
   |
43 | const _: () = cmp::const_gt!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
   |
   = note: this error originates in the macro `$crate::__::matches` which comes from the expansion of the macro `cmp::const_gt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:45:17
   |
45 | const _: bool = cmp::const_ge!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
note: required by a bound in `__AssertConstCmp`
  --> src/cmp/const_cmp.rs
   |
   | pub struct __AssertConstCmp<'a, T: ConstCmp>{
   |                                    ^^^^^^^^ required by this bound in `__AssertConstCmp`
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_ge` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): ConstCmp` is not satisfied
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:45:17
   |
45 | const _: bool = cmp::const_ge!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ConstCmp` is not implemented for `()`
   |
   = help: the following other types implement trait `ConstCmp`:
             &T
             &mut T
             BadCmp
             CmpWrapper<P>
             NonZero<i128>
             NonZero<i16>
             NonZero<i32>
             NonZero<i64>
           and $N others
   = note: required for `()` to implement `ConstCmpUnref`
note: required by a bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
  --> src/cmp/const_cmp.rs
   |
   |     R: ?Sized + ConstCmpUnref<This = T>,
   |                               ^^^^^^^^ required by this bound in `IsAConstCmp::<<T as ConstCmp>::Kind, T, R>::NEW`
...
   |     pub const NEW: Self = Self(PhantomData);
   |               --- required by a bound in this associated constant
   = note: this error originates in the macro `$crate::__coerce_to_cmp2` which comes from the expansion of the macro `cmp::const_ge` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:45:17
   |
45 | const _: bool = cmp::const_ge!(0u8, ());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `&u8`, found `&()`
   |                 arguments to this method are incorrect
   |
   = note: expected reference `&u8`
              found reference `&()`
note: method defined here
  --> src/primitive/cmp.rs
   |
   | / __declare_fns_with_docs! {
   | |     (u8, (cmp_u8))
   | |     (u16, (cmp_u16))
   | |     (u32, (cmp_u32))
...  |
   | |     macro = declare_int_cmp_fn!(),
   | | }
   | |_^
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `__declare_fns_with_docs` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:46:15
   |
46 | const _: () = cmp::const_ge!(0u8, 0u8);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
   |
   = note: this error originates in the macro `$crate::__::matches` which comes from the expansion of the macro `cmp::const_ge` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:47:17
   |
47 | const _: () = { cmp::const_ge!(BadCmp, BadCmp); };
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                 |
   |                 expected `Ordering`, found `u8`
   |                 expected due to this
   |
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_ge` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:48:15
   |
48 | const _: () = cmp::const_ge!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |               |
   |               expected `Ordering`, found `u8`
   |               expected due to this
   |
   = note: this error originates in the macro `$crate::cmp::const_cmp` which comes from the expansion of the macro `cmp::const_ge` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/cmp_macros_ui_tests/cmp_comparison_macros_type_err.rs:48:15
   |
48 | const _: () = cmp::const_ge!(BadCmp, BadCmp);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `bool`
   |
   = note: this error originates in the macro `$crate::__::matches` which comes from the expansion of the macro `cmp::const_ge` (in Nightly builds, run with -Z macro-backtrace for more info)
